teflo_datarouter_plugin/helpers.py

```diff
@@ -101,4 +94,58 @@ def send_post_req(dr_token_url, body):
         data = res.json()
         return data
     else:
-        raise TefloReportError(f'Generated access token Failed with status code {res["status_code"]}.')
+        raise TefloReportError(f'Generated access token Failed with status code {res.status_code}.')
+
+
+def get_req_status(get_url, ac_token, creds):
+    headers = {'Authorization': f'Bearer {ac_token}',
+               'X-DataRouter-Auth': 'openid-connect-client-credentials-grant',
+               }
+    req_count = 0
+    res = requests.get(url=get_url,
+                       headers=headers, verify=False)
+    if res.status_code == 401:
+        LOG.info('Unauthorized: Signature has expired, Creating new token and trying again(1/2).')
+        dr_id = creds['dr_client_id']
```

Comment:
you are using the creds, secret and dr_token_url multiple time throughout the code, can you refactor this and initialize it in the beginning, so let these values be none in the __init__ method and then in the import_artifacts method you can assign them the values coming from teflo, and then they can be class variables and be utilized throughout the code. this will avoid getting the values again and again